jwst/assign_wcs/util.py

```diff
@@ -122,47 +187,54 @@ def wcs_from_footprints(dmodels, refmodel=None, transform=None, bounding_box=Non
     """
     bb = bounding_box
     wcslist = [im.meta.wcs for im in dmodels]
+
     if not isiterable(wcslist):
         raise ValueError("Expected 'wcslist' to be an iterable of WCS objects.")
+
     if not all([isinstance(w, WCS) for w in wcslist]):
         raise TypeError("All items in wcslist are to be instances of gwcs.WCS.")
+
     if refmodel is None:
         refmodel = dmodels[0]
     else:
         if not isinstance(refmodel, DataModel):
             raise TypeError("Expected refmodel to be an instance of DataModel.")
 
-    fiducial = compute_fiducial(wcslist, bb)
+    fiducial = compute_fiducial(wcslist, bb)  # possibly retrieve from reference wcs
 
     prj = astmodels.Pix2Sky_TAN()
 
     if transform is None:
         transform = []
         wcsinfo = pointing.wcsinfo_from_model(refmodel)
         sky_axes, spec, other = gwutils.get_axes(wcsinfo)
-        rotation = astmodels.AffineTransformation2D(wcsinfo['PC'])
+
+        pc = np.reshape(
+            calc_rotation_matrix(refmodel.meta.wcsinfo.roll_ref, vparity=refmodel.meta.wcsinfo.vparity), (2, 2)
```

Comment:
`roll_ref` is in deg. Do you need to convert to rad?

Reply:
I did not! I'll fix that